scipy/stats/tests/test_quantile.py

```diff
@@ -95,21 +103,46 @@ def test_against_numpy(self, method, shape_x, shape_p, axis, xp):
 
     @pytest.mark.parametrize('axis', [0, 1])
     @pytest.mark.parametrize('keepdims', [False, True])
-    @pytest.mark.parametrize('nan_policy', ['omit', 'propagate'])
+    @pytest.mark.parametrize('nan_policy', ['omit', 'propagate', 'marray'])
     @pytest.mark.parametrize('dtype', ['float32', 'float64'])
-    def test_against_reference(self, axis, keepdims, nan_policy, dtype, xp):
+    @pytest.mark.parametrize('method', ['linear', 'harrell-davis'])
+    def test_against_reference(self, axis, keepdims, nan_policy, dtype, method, xp):
         rng = np.random.default_rng(23458924568734956)
-        shape = (3, 4)
+        shape = (5, 6)
         x = rng.random(size=shape).astype(dtype)
         p = rng.random(size=shape).astype(dtype)
-        mask = rng.random(size=shape) > 0.9
+        mask = rng.random(size=shape) > 0.8
         assert np.any(mask)
         x[mask] = np.nan
         if not keepdims:
             p = np.mean(p, axis=axis, keepdims=True)
 
+        # inject p = 0 and p = 1 to test edge cases
+        # Currently would fail with CuPy/JAX (cupy/cupy#8934, jax-ml/jax#21900)
+        if is_numpy(xp):
+            p0 = p.ravel()
+            p0[1] = 0.
+            p0[-2] = 1.
```

Comment:
consider splitting into a separate test with corresponding `@xfail_xp_backends` decorators.

Reply:
Yup, I did consider it and decided to go this way. When those linked issues are resolved, we'll just take this part out of the `if is_numpy` rather than deduplicating tests.